igvmmeasure/src/id_block.rs

```diff
@@ -39,13 +39,21 @@ pub struct SevIdBlockBuilder {
 
 impl SevIdBlockBuilder {
     pub fn build(igvm: &IgvmFile, measure: &IgvmMeasure) -> Result<Self, Box<dyn Error>> {
-        let ld = measure.digest();
         let compatibility_mask = get_compatibility_mask(igvm, IgvmPlatformType::SEV_SNP).ok_or(
             String::from("IGVM file is not compatible with the specified platform."),
         )?;
         let policy = get_policy(igvm, compatibility_mask)
             .ok_or(String::from("IGVM file does not contain a guest policy."))?;
 
+        let mut ld = [0u8; 48];
+        let digest = measure.digest();
+        if digest.len() != ld.len() {
+            return Err("Invalid digest value calculated for ID block".into());
+        }
+        for (i, val) in measure.digest().iter().enumerate() {
+            ld[i] = *val;
+        }
```

Comment:
What about using `copy_from_slice`?

```suggestion
        ld.copy_from_slice(&digest);
```

Reply:
Thanks for that. I knew there must be an easier way!